docs/architecture/pairing-adapter-for-integration-system.md

```diff
@@ -0,0 +1,29 @@
+# Pairing Adapter for Integration System
+
+## Introduction
+
+When dealing with multiple Integration Systems each of them may support different ways of establishing trusted relation for the given Application instance. To provide a consistent experience for users, we introduce the concept of Pairing Adapters. Each Integration System can be coupled with one Pairing Adapter which will abstract the details on how the Integration System handles the trusted relation prerequisites.
+
+## Solution overview
+
+This diagram illustrates the flow for pairing the Application with its External Broker using the Pairing Adapter.
+
+![External Broker Pairing Adapter](./assets/external-broker-pairing-adapter.svg)
+
+>**NOTE** The Application must be created before the pairing is performed. To create an Application that is managed by the Integration System the Application should be created using Application Template.
+
+### Steps
+
+1. The Administrator requests the pairing data for the Application using the Cockpit (1).
+
+1. The Cockpit contacts the Director component to obtain the pairing data for the Application (2). The Application is loosely coupled with the Integration System that manages it. Thanks to this relation the Director component is aware which Pairing Adapter is assigned to the Integration System. Knowing this, the Pairing Adapter is called to request the pairing data (3).
+
+1. The Pairing Adapter calls External Broker to obtain pairing data (4).
+
+1. Administrator after obtaining the pairing data provides them to the Application instance which allows the Application instance to establish trusted relation with External Broker (6). This informs External Broker that there is a new Application instance that needs to be managed via this External Broker.
```

Comment:
Something is wrong here, step 4 says about point 5. Please take a look.

Reply:
The numbers reflecting the numbers from the diagram are in between parenthesis.